src/app/features/automated-status-update/automated-field-mapping/automated-field-mapping.component.ts

```diff
@@ -176,12 +176,17 @@ export class AutomatedFieldMappingComponent implements OnInit {
     });
     if (this.isInvalid) return;
 
-    const formattedMappings: { [key: string]: any } = {};
-    Object.entries(this.selectedMappings).forEach(([key, value]) => {
-      formattedMappings[key] = this.schemaService.valueToDatabaseFormat(
-        value,
-        this.targetFieldConfig,
-      );
+    const formattedMappings: { [key: string]: any } = {
+      ...this.selectedMappings,
```

Comment:
as the forms are initialized with the values from selectedMappings, this shouldn't really be necessary?